src/codegen.cpp

```diff
@@ -3450,6 +3451,29 @@ static void finalize_gc_frame(Function *F)
         ptlsStates->setCalledFunction(getter);
         ptlsStates->setAttributes(jltls_states_func->getAttributes());
     }
+    else if (jl_tls_offset != -1) {
+#ifdef LLVM37
+        const char *asm_str = nullptr;
+#  if defined(_CPU_X86_64_)
+        asm_str = "movq %fs:0, $0";
+#  elif defined(_CPU_X86_)
+        asm_str = "movl %gs:0, $0";
+#  elif defined(_CPU_AARCH64_)
+        asm_str = "mrs $0, tpidr_el0";
+#  endif
```

Comment:
Yeah. (Although we resolve `jl_get_ptls_states` to the real one in codegen.) I didn't think about this because this is documented in https://github.com/JuliaLang/julia/blob/bf44142940b528861d0925db776cabfffc2ead51/src/codegen.cpp#L5136